engines/engine.go

```diff
@@ -1,6 +1,20 @@
-package engine
+package engines
 
-import "github.com/taskcluster/taskcluster-worker/runtime"
+import (
+	"github.com/taskcluster/taskcluster-worker/runtime"
+	"github.com/taskcluster/taskcluster-worker/runtime/gc"
+)
+
+// An EngineOptions is a wrapper for the set of options/arguments given when
+// an Engine is created.
+//
+// We pass all options as a single argument, so that we can add additional
+// properties without breaking source compatibility.
+type EngineOptions struct {
+	GarbageCollector *gc.GarbageCollector
+	//TODO: Add some sort of interface to the system logger
+	//TODO: Add some interface to submit statistics for influxdb/signalfx
```

Comment:
I think `runtime.Environment` can be a property on EngineOptions.. .And since this is strictly for wrapping arguments to `EngineProvider` I suggest we move it into the extpoints package which is the only place it is used...